### PR TITLE
Improved index-sync.sh script, added Index Recovery steps to docs

### PR DESCRIPTION
**Pull request by opopovss**

### CODE REVIEW

File: app/util/jira/index-sync.sh [hunk 1]
```diff
@@ -35,6 +36,42 @@ function find_word_in_log() {
             exit 1
         fi
 }
+# Check if is correct index snapshot for Jira DC is generated
+function find_correct_snapshot() {
+        SLEEP_TIME=100
```
[the remark this answers is not in the document]
Comment (opopovss) [reply app/util/jira/index-sync.sh:41]: To avoid spamming a message about checking the size of a snapshot

**Review by dmika1 — APPROVED** [review]

File: app/util/jira/index-sync.sh [hunk 2]
```diff
@@ -35,6 +43,56 @@ function find_word_in_log() {
...
+        ATTEMPTS=$((TIMEOUT / SLEEP_TIME))
+        while [ ${COUNTER} -lt ${ATTEMPTS} ];do
+            # Get the latest snapshot from the index-snapshots folder
+            # shellcheck disable=SC2031
+            if [[ ${JIRA_VERSION} == 8 ]]; then
+              SNAPSHOT=$(sudo su -c "ls -tr /media/atl/jira/shared/export/indexsnapshots/IndexSnapshot*" 2>/dev/null | tail -1)
+            elif [[ ${JIRA_VERSION} == 9 ]]; then
+              SNAPSHOT=$(sudo su -c "ls -tr /media/atl/jira/shared/caches/indexesV2/snapshots/IndexSnapshot*" 2>/dev/null | tail -1)
+            fi
+            if  sudo su -c "test -z ${SNAPSHOT}"; then
```
Comment (ometelytsia) [inline app/util/jira/index-sync.sh:59]:
```
JIRA_8_SNAPSHOT_PATH="/media/atl/jira/shared/export/indexsnapshots/IndexSnapshot*"
JIRA_9_SNAPSHOT_PATH="/media/atl/jira/shared/caches/indexesV2/snapshots/IndexSnapshot*"

if [[ ${JIRA_VERSION} == 8 ]]; then
  JIRA_SNAPSHOT_PATH=$JIRA_8_SNAPSHOT_PATH
elif [[ ${JIRA_VERSION} == 9 ]]; then
  JIRA_SNAPSHOT_PATH=$JIRA_9_SNAPSHOT_PATH
else
  echo "ERROR: unsupported major Jira version: $JIRA_VERSION"
  exit 1
fi

SNAPSHOT=$(sudo su -c "ls -tr $JIRA_SNAPSHOT_PATH" 2>/dev/null | tail -1)
```

File: docs/dc-apps-performance-toolkit-user-guide-jira.md [hunk 3]
```diff
@@ -645,6 +645,47 @@ For more information, go to [Re-indexing Jira](https://confluence.atlassian.com/
...
+    export BASTION_IP=bastion_instance_public_ip
+    export NODE_IP=node_private_ip
+    export SSH_OPTS1='-o ServerAliveInterval=60'
+    export SSH_OPTS2='-o ServerAliveCountMax=30'
+    ssh ${SSH_OPTS1} ${SSH_OPTS2} -o "proxycommand ssh -W %h:%p ${SSH_OPTS1} ${SSH_OPTS2} ec2-user@${BASTION_IP}" ec2-user@${NODE_IP}
+    ```
+2. Once you're in the node, run command corresponding to your Jira version:
+   
+   
+   **Jira 9.0.0**
```
Comment (ometelytsia) [inline docs/dc-apps-performance-toolkit-user-guide-jira.md:667]: Jira 9

File: docs/dc-apps-performance-toolkit-user-guide-jira.md [hunk 4]
```diff
@@ -645,6 +645,47 @@ For more information, go to [Re-indexing Jira](https://confluence.atlassian.com/
...
+   ```bash
+    sudo su -c "du -sh  /media/atl/jira/shared/caches/indexesV2/snapshots/IndexSnapshot*" | tail -1
+   ```
+   **Jira 8.20.13**
```
Comment (ometelytsia) [inline docs/dc-apps-performance-toolkit-user-guide-jira.md:671]: Jira 8

File: docs/dc-apps-performance-toolkit-user-guide-jira.md [hunk 5]
```diff
@@ -645,6 +645,48 @@ For more information, go to [Re-indexing Jira](https://confluence.atlassian.com/
 
 Jira will be unavailable for some time during the re-indexing process. When finished, the **Acknowledge** button will be available on the re-indexing page.
 
+---
+
+#### <a id="indexrecovery"></a> Index Recovery (~15 min, only for Jira versions below 9.1.0)
```
Comment (ometelytsia) [inline docs/dc-apps-performance-toolkit-user-guide-jira.md:650]: (~15 min, only for Jira versions 9.0.x and below. For Jira 9.1.0+ skip this step.)

File: docs/dc-apps-performance-toolkit-user-guide-jira.md [hunk 6]
```diff
@@ -645,6 +645,48 @@ For more information, go to [Re-indexing Jira](https://confluence.atlassian.com/
...
+
+1. Log in as a user with the **Jira System Administrators** [global permission](https://confluence.atlassian.com/adminjiraserver/managing-global-permissions-938847142.html).
+2. Go to **![cog icon](/platform/marketplace/images/cog.png) &gt; System &gt; Indexing**.
+3. In the **Index Recovery** click **Edit Settings**.
+4. Set the recovery index schedule to 5min ahead of the current server time.
+5. Wait ~10min until the index snapshot is created.
+
+Jira will be unavailable for some time during the index recovery process.
```
Comment (ometelytsia) [inline docs/dc-apps-performance-toolkit-user-guide-jira.md:658]: Does 1st node unavailable during the index recovery creation process?

File: docs/dc-apps-performance-toolkit-user-guide-jira.md [hunk 7]
```diff
@@ -880,6 +899,14 @@ To receive scalability benchmark results for two-node Jira DC **with** app-speci
    docker run --shm-size=4g -v "$PWD:/dc-app-performance-toolkit" atlassian/dcapt jira.yml
    ```
 
+
+   {{% /warning %}}
```
Comment (ometelytsia) [inline docs/dc-apps-performance-toolkit-user-guide-jira.md:903]: Let's move this warning after step 6.

File: docs/dc-apps-performance-toolkit-user-guide-jsm.md [hunk 8]
```diff
@@ -802,6 +802,48 @@ For more information, go to [Re-indexing Jira](https://confluence.atlassian.com/
 
 Jira Service Management will be unavailable for some time during the re-indexing process. When finished, the **Acknowledge** button will be available on the re-indexing page.
 
+---
+
+#### <a id="indexrecovery"></a> Index Recovery (~15 min, only for Jira Service Management versions below 5.1.0)
```
Comment (ometelytsia) [inline docs/dc-apps-performance-toolkit-user-guide-jsm.md:807]: The same rewording as in Jira

File: docs/dc-apps-performance-toolkit-user-guide-jira.md (hunk 7 above)
Comment (ometelytsia) [reply docs/dc-apps-performance-toolkit-user-guide-jira.md:903]: Wrong "/" in the warning. Pls fix.

File: docs/dc-apps-performance-toolkit-user-guide-jsm.md [hunk 9]
```diff
@@ -1009,42 +1051,22 @@ To receive scalability benchmark results for two-node Jira Service Management DC
...
+6. Go to **![cog icon](/platform/marketplace/images/cog.png) &gt; System &gt; Clustering** and check there is expected number of nodes with node status `ACTIVE` and application status `RUNNING`. To make sure that Jira Service Management index successfully synchronized to the second node.
+7. Run toolkit with docker from the execution environment instance:
 
    ``` bash
    cd dc-app-performance-toolkit
    docker pull atlassian/dcapt
    docker run --shm-size=4g -v "$PWD:/dc-app-performance-toolkit" atlassian/dcapt jsm.yml
    ```
+   
+   {{% /warning %}}
```
Comment (ometelytsia) [inline docs/dc-apps-performance-toolkit-user-guide-jsm.md:1064]: Let's move afrer step 6.

Comment (ometelytsia) [reply docs/dc-apps-performance-toolkit-user-guide-jsm.md:1064]: Wrong "/" in warning.

Comment (ometelytsia) [reply docs/dc-apps-performance-toolkit-user-guide-jsm.md:1064]: Pls use `atlas dac watch` to preview syntax.

**Review by ometelytsia — APPROVED** [review]

**Pull request merged**